package/MDAnalysis/lib/nsgrid.pyx

```diff
@@ -0,0 +1,943 @@
+                                    if dd == 0:
+                                        shift = i
+                                    elif dd == 1:
+                                        shift = j
+                                    else:
+                                        shift = k
+
+                                    if shift:
+                                        d2new_c = 0
+
+                                        for d in range(DIM):
+                                            d2new_c += (pos[d] + trial[d] - shift*box[dd, d])**2
+
+                                        if d2new_c <= BOX_MARGIN*d2new:
+                                            use = False
+
+                                if use: # Accept this shift vector.
+                                    if self.c_pbcbox.ntric_vec >= MAX_NTRICVEC:
+                                        with gil:
+                                            print("\nWARNING: Found more than %d triclinic "
+                                                  "correction vectors, ignoring some."
+                                                  % MAX_NTRICVEC)
+                                            print("  There is probably something wrong with "
+                                                  "your box.")
+                                            print(np.array(box))
+
+                                            for i in range(self.c_pbcbox.ntric_vec):
+                                                print(" -> shift #{}: [{}, {}, {}]".format(i+1,
+                                                                                           self.c_pbcbox.tric_shift[i][XX],
+                                                                                           self.c_pbcbox.tric_shift[i][YY],
+                                                                                           self.c_pbcbox.tric_shift[i][ZZ]))
+                                    else:
+                                        for d in range(DIM):
+                                            self.c_pbcbox.tric_vec[self.c_pbcbox.ntric_vec][d] = \
+                                                trial[d]
+                                        self.c_pbcbox.tric_shift[self.c_pbcbox.ntric_vec][XX] = i
+                                        self.c_pbcbox.tric_shift[self.c_pbcbox.ntric_vec][YY] = j
+                                        self.c_pbcbox.tric_shift[self.c_pbcbox.ntric_vec][ZZ] = k
+                                        self.c_pbcbox.ntric_vec += 1
+
+
+    def update(self, real[:,::1] box):
+        if box.shape[0] != DIM or box.shape[1] != DIM:
+            raise ValueError("Box must be a %i x %i matrix. (shape: %i x %i)" %
+                             (DIM, DIM, box.shape[0], box.shape[1]))
+        if (box[XX, XX] == 0) or (box[YY, YY] == 0) or (box[ZZ, ZZ] == 0):
+            raise ValueError("Box does not correspond to PBC=xyz")

```

Comment:
As the box comes from the Universe instance, it can see three cases where this means something:
1. The trajectory file is corrupted
2. There is a bug in the box detection algorithm (when the trajectory is read).
3. Someone actually wants to perform NS on a system where PBC=xy for instance (in which case the grid-based NS would not work)

Reply:
In Gromacs you can do `pbc=xy` simulations. I don't know what the box looks like in this case. 

Ok, let's check for it then. I'd change the error message to "unit cell box contains collinear or null box vectors: NS only works in 3D"